lib/StripeResource.js

```diff
@@ -282,15 +330,57 @@ StripeResource.prototype = {
         path: path,
       });
 
+      var requestRetries = numRetries || 0;
+
       req._requestEvent = requestEvent;
 
       req._requestStart = Date.now();
 
       self._stripe._emitter.emit('request', requestEvent);
 
       req.setTimeout(timeout, self._timeoutHandler(timeout, req, callback));
-      req.on('response', self._responseHandler(req, callback));
-      req.on('error', self._errorHandler(req, callback));
+
+      req.on('response', function(res) {
```

Comment:
This function and the one for `req.on('error' ...` share a lot of code. Might want to combine them into one functions that's reusable.